core/trino-main/src/test/java/io/trino/metadata/CountingMetadataManagerWrapper.java

```diff
@@ -0,0 +1,860 @@
+import io.trino.Session;
+import io.trino.connector.CatalogName;
+import io.trino.spi.connector.AggregateFunction;
+import io.trino.spi.connector.AggregationApplicationResult;
+import io.trino.spi.connector.BeginTableExecuteResult;
+import io.trino.spi.connector.CatalogSchemaName;
+import io.trino.spi.connector.CatalogSchemaTableName;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorCapabilities;
+import io.trino.spi.connector.ConnectorOutputMetadata;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.Constraint;
+import io.trino.spi.connector.ConstraintApplicationResult;
+import io.trino.spi.connector.JoinApplicationResult;
+import io.trino.spi.connector.JoinCondition;
+import io.trino.spi.connector.JoinStatistics;
+import io.trino.spi.connector.JoinType;
+import io.trino.spi.connector.LimitApplicationResult;
+import io.trino.spi.connector.MaterializedViewFreshness;
+import io.trino.spi.connector.ProjectionApplicationResult;
+import io.trino.spi.connector.SampleApplicationResult;
+import io.trino.spi.connector.SampleType;
+import io.trino.spi.connector.SortItem;
+import io.trino.spi.connector.SystemTable;
+import io.trino.spi.connector.TableColumnsMetadata;
+import io.trino.spi.connector.TableScanRedirectApplicationResult;
+import io.trino.spi.connector.TopNApplicationResult;
+import io.trino.spi.expression.ConnectorExpression;
+import io.trino.spi.function.OperatorType;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.security.GrantInfo;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.Privilege;
+import io.trino.spi.security.RoleGrant;
+import io.trino.spi.security.TrinoPrincipal;
+import io.trino.spi.statistics.ComputedStatistics;
+import io.trino.spi.statistics.TableStatistics;
+import io.trino.spi.statistics.TableStatisticsMetadata;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeManager;
+import io.trino.sql.analyzer.TypeSignatureProvider;
+import io.trino.sql.planner.PartitioningHandle;
+import io.trino.sql.tree.QualifiedName;
+import io.trino.transaction.TransactionManager;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.Set;
+
+public class CountingMetadataManagerWrapper
+        implements Metadata
+{
+    private final Metadata delegate;
+    // TODO: Not every method's call counting is implemented. It will be implemented based on testing needs.
+    private long getTableStatisticsCallCount;
+
+    public CountingMetadataManagerWrapper(
+            FeaturesConfig featuresConfig,
+            SystemSecurityMetadata systemSecurityMetadata,
+            TransactionManager transactionManager,
+            GlobalFunctionCatalog globalFunctionCatalog,
+            TypeManager typeManager)
+    {
+        delegate = new MetadataManager(featuresConfig,
```

Comment:
nit: per code style, `featuresConfig` should be on the next line

however, `CountingMetadataManagerWrapper`  should actually take `delegate` as an argument